### PR TITLE
Support Digit and Decimal values in CharUnicodeInfo

### PR DESCRIPTION
**Pull request by tarekgh**
This change adding the needed data for the digit and decimal values and the needed code to use the data and return the required values

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @jkotas @weshaggard @danmosemsft

File: src/mscorlib/corefx/System/Globalization/CharUnicodeInfoData.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public static partial class CharUnicodeInfo
 
         // 12:4:4 index table of the Unicode cateogry data.
         private static ushort[] s_pCategoryLevel1Index = new ushort[]
-        {
+        {
```
Comment (jkotas) [inline src/mscorlib/corefx/System/Globalization/CharUnicodeInfoData.cs]: Nit - extra white spaces

File: src/mscorlib/corefx/System/Globalization/CharUnicodeInfo.cs [hunk 2]
```diff
@@ -191,6 +191,22 @@ internal unsafe static double InternalGetNumericValue(int ch)
             }
         }
 
+        internal unsafe static ushort InternalGetDigitValues(int ch)
+        {
+            Contract.Assert(ch >= 0 && ch <= 0x10ffff, "ch is not in valid Unicode range.");
+            // Get the level 2 item from the highest 12 bit (8 - 19) of ch.
+            ushort index = s_pNumericLevel1Index[ch >> 8];
+            // Get the level 2 WORD offset from the 4 - 7 bit of ch.  This provides the base offset of the level 3 table.
+            // The offset is referred to an float item in m_pNumericFloatData.
```
Comment (jkotas) [inline src/mscorlib/corefx/System/Globalization/CharUnicodeInfo.cs]: Why does this talk about `m_pNumericFloatData`? The comment seems outdated.

Comment (tarekgh) [inline src/mscorlib/corefx/System/Globalization/CharUnicodeInfo.cs]: I'll fix the comment. thanks for the catch

**Comment by tarekgh** [comment]
thanks @jkotas

**Pull request merged**